pkg/store/bucket.go

```diff
@@ -2104,6 +2122,15 @@ func (r *bucketIndexReader) LoadSeriesForTime(ref uint64, lset *[]symbolizedLabe
 // Close released the underlying resources of the reader.
 func (r *bucketIndexReader) Close() error {
 	r.block.pendingReaders.Done()
+
+	r.block.valuesMutex.Lock()
+	for k, v := range r.valueSymbols {
+		// Only update the cache if the block cache entry is not set.
+		if v != zeroIndexSymbol && r.block.valueSymbols[k] == zeroIndexSymbol {
+			r.block.valueSymbols[k] = v
+		}
```

Comment:
This might be a problem. Not sure which way is better. Overwrite the entry every time?